LPFilter/Source/PluginEditor.cpp

```diff
@@ -41,10 +63,47 @@ void LpfilterAudioProcessorEditor::buttonClicked(Button* button)
     // In case there is a second button, we must check wich of the buttons called this method
     if (button == &bypassButton)
     {
-        *processor.bypassParam =(button->getToggleState()) ;
+        *processor.bypassParam = button->getToggleState() ;
     }
 }
 
+void LpfilterAudioProcessorEditor::comboBoxChanged(ComboBox* comboBox)
+{
+    // Update parameters from UI
+    
+    if (comboBox == &filterModeBox)
+    {
+        int index = filterModeBox.getSelectedItemIndex(); // values = 0 ,1, 2
+        
+        *processor.mode = index;
+        
+    }
+}
+
+void LpfilterAudioProcessorEditor::sliderDragStarted (Slider* slider)
+{
+    if (slider == &frequencyKnob)
+        processor.frequency->beginChangeGesture();
```

Comment:
You should do the same thing for all UI components that change an automatable parameter.
So, you should 
- add mouse listener to `filterModeBox` and `bypassButton`
- override `mouseDown` and mouseUp` methods 
- call `beginChangeGesture` and `endChangeGesture` accordingly